packages/client/src/links/wsLink.ts

```diff
@@ -28,6 +28,8 @@ export interface WebSocketClientOptions {
   retryDelayMs?: typeof retryDelay;
   onOpen?: () => void;
   onClose?: (cause?: { code?: number }) => void;
+  lazy?: boolean;
+  disconnectDelayMs?:number;
```

Comment:
Make this discriminated, something like this:

```suggestion
  lazy?: {
    enabled: false;
  } | {
     enabled: true;
     disconnectDelayMs?:number;
  }
```